zipkin-lens/src/components/App/SidebarMenu.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2015-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  createStyles,
+  makeStyles,
+  ListItem,
+  ListItemProps,
+  Theme,
+  Tooltip,
+} from '@material-ui/core';
+import classNames from 'classnames';
+import React from 'react';
+import { Link, useLocation } from 'react-router-dom';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    item: {
+      height: '3.2rem',
+      cursor: 'pointer',
+      fontSize: '1.05rem',
+      color: theme.palette.grey[400],
+      '&:hover': {
+        color: theme.palette.common.white,
+        backgroundColor: theme.palette.primary.dark,
+      },
+    },
+    'item--selected': {
+      color: theme.palette.common.white,
+      backgroundColor: theme.palette.primary.dark,
+    },
+  }),
+);
+
+interface SidebarMenuProps extends React.DOMAttributes<HTMLElement> {
+  title: string;
+  path: string;
+  icon: IconProp;
+}
+
+export const SidebarMenuImpl = React.forwardRef<
```

Comment:
just curious.
why is this component made forwardRef?

https://github.com/openzipkin/zipkin/commit/7a1790f3b3e61110cd4e7263219702d398b64efc#diff-d7507c88e4d135f78935a54009f7fefd

Reply:
Needed to pass a ref to the DOM element to position the popup menu

https://github.com/openzipkin/zipkin/blob/master/zipkin-lens/src/components/App/LanguageSelector.jsx#L85

Pretty sure I tried without ref and wasn't being able to position the popup menu well, let me know if you have any alternate ideas though.

Reply:
I get it! thanks